src/main/java/com/stripe/model/v2/EventNotification.java

```diff
@@ -0,0 +1,103 @@
+package com.stripe.model.v2;
+
+import com.google.gson.JsonObject;
+import com.google.gson.annotations.SerializedName;
+import com.stripe.StripeClient;
+import com.stripe.exception.StripeException;
+import com.stripe.model.StripeObject;
+import com.stripe.model.v2.Event.RelatedObject;
+import com.stripe.net.ApiMode;
+import com.stripe.net.ApiResource;
+import com.stripe.net.ApiResource.RequestMethod;
+import com.stripe.net.RawRequestOptions;
+import com.stripe.net.StripeResponse;
+import java.time.Instant;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+/**
+ * `EventNotification` represents the common properties for json that's delivered from an Event
+ * Destination. A concrete child of `EventNotification` will be returned from
+ * `StripeClient.parseEventNotificaion()`. You will likely want to cast that object to a more
+ * specific child of `EventNotification`, like `PushedV1BillingMeterErrorReportTriggeredEvent`
+ */
+@Getter
+public abstract class EventNotification {
+
+  /** Unique identifier for the event. */
+  @SerializedName("id")
+  public String id;
+
+  /** The type of the event. */
+  @SerializedName("type")
+  public String type;
+
+  /** Time at which the object was created. */
+  @SerializedName("created")
+  public Instant created;
+
+  /** Livemode indicates if the event is from a production(true) or test(false) account. */
+  @SerializedName("livemode")
+  public Boolean livemode;
+
+  /** [Optional] Authentication context needed to fetch the event or related object. */
+  @SerializedName("context")
+  public String context;
+
+  /** [Optional] Reason for the event. */
+  @SerializedName("reason")
+  public com.stripe.model.v2.Event.Reason reason;
+
+  @Getter(AccessLevel.NONE)
+  protected transient StripeClient client;
+
+  /**
+   * Helper for constructing an Event Notification. Not recommended for production code (because it
+   * lacks signature validation) but useful in testing. StripeClient uses it under the hood.
+   */
+  public static EventNotification fromJson(String payload, StripeClient client) {
```

Comment:
this is new

Reply:
same comment as go - users might like to use it for their own tests. But we can start it protected and open it up later, if we want

Reply:
also it looks like we cant use it in tests if it's private. The more I think about it though, the more broadly useful it is. Like, if someone validates a webhook and then saves the string to a database. when they pull it out later, they'll want the full `EventNotification`, but don't need to validate it

Reply:
unfortunately, the client and the model are different packages, so it yells. public it is!